networkx/linalg/laplacianmatrix.py

```diff
@@ -146,6 +147,38 @@ def normalized_laplacian_matrix(G, nodelist=None, weight="weight"):
     return sp.sparse.csr_matrix(DH @ (L @ DH))
 
 
+def total_spanning_tree_weight(G, weight=None):
+    """
+    Returns the total weight of all spanning trees of `G`.
+
+    Kirchoff's Tree Matrix Theorem states that the determinant of any cofactor of the
+    Laplacian matrix of a graph is the number of spanning trees in the graph. For a
+    weighted Laplacian matrix, it is the sum across all spanning trees of the
+    multiplicative weight of each tree. That is, the weight of each tree is the
+    product of its edge weights.
+
+    Parameters
+    ----------
+    G : NetworkX Graph
+        The graph to use Kirchhoff's theorem on.
+
+    weight : string or None
+        The key for the edge attribute holding the edge weight. If `None`, then
+        each edge is assumed to have a weight of 1 and this function returns the
+        total number of spanning trees in `G`.
+
+    Returns
+    -------
+    float
+        The sum of the total multiplicative weights for all spanning trees in `G`
+    """
+    import numpy as np
+
+    G_laplacian = nx.laplacian_matrix(G, weight=weight).toarray()
+    # Determinant ignoring first row and column
+    return abs(np.linalg.det(G_laplacian[1:, 1:]))
```

Comment:
As a future improvement, it'd be nice to keep the determinant computation sparse (`laplacian_matrix` already depends on scipy, so making this function also require scipy wouldn't expand the dependency footprint). However there is no `scipy.sparse.linalg.det`, so you'd have to implement it yourself which would detract from readability.

Adding a `det` method to `scipy.sparse.linalg` seems like a potentially interesting idea though...

Reply:
I didn't actually even notice this. Within the scope of a tsp problem which is what this was originally written for, all of the graphs are complete so I explicitly wanted to be working with ndarrays since the sparse arrays provide no benefit and as you mentioned `scipy.sparse.linalg` has no `det`. I'm not really interested in writing one here, as it probably should go into scipy anyways. Is there any way to that advantage of the sparse nature of the data for calculating the determinant? Obviously we don't what to enumerate the entire matrix if possible but I'm not sure how all of the sparse data structures work.